src/main/java/eu/xlime/kafka/rdf/BaseXLiMeResourceToMongo.java

```diff
@@ -92,7 +92,32 @@ private String summariseResourcesProcessed() {
 		
 		return sb.toString();
 	}
+	
+	protected Summary generateObjectSummary(){
+		Summary sum = new Summary();
+		sum.setConsumerId(getClass().getSimpleName() + "_" + instId);
+		sum.setProcessed(getMessagesProcessed());
+		sum.setAnnotations(summariseResourcesProcessedForObject());
+		return sum;
+	}
 
+	private Map<String,Long> summariseResourcesProcessedForObject() {

```

Comment:
Looks good, To avoid code duplication (this method repeats a lot of what's in summariseResourcesProcessed), remove summariseResourcesProcessed and change generateSummary() to use the output of generateObjectSummary().